### PR TITLE
dcgm: 3.3.5 -> 3.3.9; cudaPackages_10{,_0,_1,_2}: drop

### PR DESCRIPTION
**Pull request by emilazy**
An update of DCGM to a newer version that has dropped CUDA 10 support upstream, fixing the build and cleaning it up a bit in the process, and a [scheduled visit from the reaper](https://github.com/NixOS/nixpkgs/pull/342112) now that 24.11 has branched off and this is the last thing standing in the way of my unsupported‐compiler‐removing rampage. See the commit messages for details. CUDA 11 lives for now, mostly pending action on https://github.com/NixOS/nixpkgs/pull/345658, though it’s not urgent as there are other blockers to removing the slightly newer GCCs that I have to deal with first.

I don’t use DCGM, so cc @de11n for testing. It passes tests that we weren’t running before, at least.

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 12 packages built:</summary>
  <ul>
    <li>caffe</li>
    <li>caffe.bin</li>
    <li>cudaPackages.cudatoolkit-legacy-runfile</li>
    <li>cudaPackages.cudatoolkit-legacy-runfile.doc</li>
    <li>cudaPackages.cudatoolkit-legacy-runfile.lib</li>
    <li>cudaPackages_11.cudatoolkit-legacy-runfile</li>
    <li>cudaPackages_11.cudatoolkit-legacy-runfile.doc</li>
    <li>cudaPackages_11.cudatoolkit-legacy-runfile.lib</li>
    <li>dcgm</li>
    <li>prometheus-dcgm-exporter</li>
    <li>python311Packages.caffe</li>
    <li>python311Packages.caffe.bin</li>
  </ul>
</details>

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by de11n — APPROVED** [review]
Wow, thank you for cleaning up and fixing DCGM. I had poked at it a while ago and didn't have time to push it forward. This is a great improvement.

File: pkgs/by-name/dc/dcgm/package.nix [hunk 1]
```diff
@@ -54,44 +39,43 @@ let
...
+      (lib.cmakeFeature "CUDA${version}_LIBS" "${cudaPackages.cuda_cudart.stubs}/lib/stubs/libcuda.so")
+      (lib.cmakeFeature "CUDA${version}_STATIC_LIBS" "${lib.getLib cudaPackages.cuda_cudart}/lib/libcudart.so")
+      (lib.cmakeFeature "CUDA${version}_STATIC_CUBLAS_LIBS" (lib.concatStringsSep ";" [
+        "${lib.getLib cudaPackages.libcublas}/lib/libcublas.so"
+        "${lib.getLib cudaPackages.libcublas}/lib/libcublasLt.so"
+      ]))
+    ];
 
 # gcc11 is required by DCGM's very particular build system
 # C.f. https://github.com/NVIDIA/DCGM/blob/7e1012302679e4bb7496483b32dcffb56e528c92/dcgmbuild/build.sh#L22
```
Comment (de11n) [inline pkgs/by-name/dc/dcgm/package.nix:63]: I guess `gcc11` is no longer strictly required?

Comment (emilazy) [reply pkgs/by-name/dc/dcgm/package.nix]: Ah, yes, I just forgot to drop this comment; it’s building with GCC 13 now. Fixed :)

**Review by de11n — APPROVED** [review]

**Comment by emilazy** [comment]
Rebased for conflicts.

**Review by de11n — APPROVED** [review]

**Review by ConnorBaker — APPROVED** [review]
Thank you for taking this on!

**Pull request merged**